### PR TITLE
default add progress to false, unless using CLI

### PR DESCRIPTION
**Pull request by whyrusleeping**
Should fix issues seen in https://github.com/ipfs/js-ipfs-api/issues/323

License: MIT
Signed-off-by: Jeromy <why@ipfs.io>

### CODE REVIEW

File: core/commands/add.go [hunk 1]
```diff
@@ -81,6 +81,11 @@ You can now refer to the added file in a gateway, like so:
 			return nil
 		}
 
+		_, found, _ := req.Option(progressOptionName).Bool()
```
Comment (Kubuxu) [inline core/commands/add.go]: That seems quite cheaty.
Maybe we should just introduce something like `request agent`? to the `Request` object.

Comment (whyrusleeping) [inline core/commands/add.go]: That might be a good idea, except it changes the http api. which i'm explicitly trying to avoid right now.

Comment (Kubuxu) [inline core/commands/add.go]: I have missed that the code is in PreRun function.

Everything is much less hacky now.

**Comment by Kubuxu** [comment]
LGTM

**Pull request merged**